docs/tuning-apps/networking/cuda.rst

```diff
@@ -4,51 +4,96 @@ CUDA
 
-   This is the preferred mechanism.  Since UCX will be providing the
-   CUDA support, it is important to ensure that UCX itself is built
-   with CUDA support.
+What kind of CUDA support exists in Open MPI?
+---------------------------------------------
 
-   To see if your ucx was built with CUDA support run the following
-   command:
+Open MPI depends on various features of CUDA 4.0, so one needs to have
+at least the CUDA 4.0 driver and toolkit.  The new features of
+interest are the Unified Virtual Addressing (UVA) so that all pointers
+within a program have unique addresses.  In addition, there is a new
+API that allows one to determine if a pointer is a CUDA device pointer
+or host memory pointer.  This API is used by the library to decide
+what needs to be done with each buffer.  In addition, CUDA 4.1 also
+provides the ability to register host memory with the CUDA driver,
+which can improve performance.  CUDA 4.1 also added CUDA IPC support
+for fast communication between GPUs on the same node.
 
-   .. code-block:: sh
+.. error:: CUDA 4.0 is SUPER OLD!  End users dont care about the
+   differences between cuda-aware, cuda-ipc, gpu-direct, and gpu-direct-rdma
 
-      # Check if ucx was built with CUDA support
-      shell$ ucx_info -v
+Note that derived datatypes |mdash| both contiguous and non-contiguous
+|mdash| are supported.  However, the non-contiguous datatypes
+currently have high overhead because of the many calls to the CUDA
+function ``cuMemcpy()`` to copy all the pieces of the buffer into the
+intermediate buffer.
 
-      # configured with: --build=powerpc64le-redhat-linux-gnu --host=powerpc64le-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-optimizations --disable-logging --disable-debug --disable-assertions --enable-mt --disable-params-check --enable-cma --without-cuda --without-gdrcopy --with-verbs --with-cm --with-knem --with-rdmacm --without-rocm --without-xpmem --without-ugni --without-java
+CUDA-aware support is available in:
 
-   If you need to build ucx yourself to include CUDA support, please
-   see the UCX documentation for `building ucx with Open MPI: <https://openucx.readthedocs.io/en/master/running.html#openmpi-with-ucx>`_
+* The UCX (``ucx``) PML
+* The PSM2 (``psm2``) MTL with the CM (``cm``) PML.
+* The OFI (``ofi``) MTL with the CM (``cm``) PML.
+* Both CUDA-ized shared memory (``smcuda``) and TCP (``tcp``) BTLs
+  with the OB1 (``ob1``) PML.
+* The HCOLL (``hcoll``) COLL
 
-   It should look something like:
+See :ref:`this FAQ entry <faq-cuda-mpi-apis-cuda-label>`
+for more details on which commnication APIs are CUDA-aware.
 
-   .. code-block:: sh
+OFI support for CUDA
```

Comment:
Might want to add something about versions of libfabric to use.  Maybe say something like latest version of libfabric is recommended or at least something newer than 1.9?